wgpu-hal/src/metal/mod.rs

```diff
@@ -12,6 +12,9 @@ resources, followed by other bind groups. The vertex buffers are bound at the ve
 end of the VS buffer table.
 
 !*/
+// Avoid noise, many objc2-metal functions are still `unsafe`.
+// See also https://github.com/madsmtm/objc2/issues/685.
+#![allow(unsafe_op_in_unsafe_fn)]
```

Comment:
thought: IMO, it's okay to have this noise, unless you're _really_ convinced that most of it will fall away. 🙂

Reply:
To this end, we actually put this lint on intentionally and migrated all of hal to use it before it was enabled by default, so we are quite happy with having unsafe around.

Reply:
I am fairly confident that a lot of the `unsafe` will disappear, and so I felt that adding that line was valuable, since it made the already large diff easier to review.

I guess adding `unsafe` now would make it easier for you to migrate once the things are marked safe? Another option would be to open a new tracking issue and assigning me to it.

Regardless, I'll try to prioritize this work in `objc2`.

Reply:
After https://github.com/madsmtm/objc2/pull/792, a lot of it did fall away, though unfortunately not as much as I'd have liked, we're still working on that in https://github.com/madsmtm/objc2/issues/793. I have removed this `#![allow(unsafe_op_in_unsafe_fn)]` now though, and made the unsafe blocks more local instead.